lib/ipfs/IpfsStorage.ts

```diff
@@ -44,8 +42,48 @@ export default class IpfsStorage {
     return IpfsStorage.ipfsStorageSingleton;
   }
 
-  private constructor (node: IPFS) {
+  private constructor (node: IPFS, repo: any) {
+    this.repo = repo;
     this.node = node;
+    this.healthy = true;
+    this.healthCheckInternalInSeconds = 60;
+  }
+
+  private static async getNode (repo?: any): Promise<IPFS> {
+    const localRepoName = 'sidetree-ipfs';
+    const options = {
+      repo: repo !== undefined ? repo : localRepoName
+    };
+    const node = await IPFS.create(options);
+    return node;
+  }
+
+  /**
+   * performs health check once every minute
+   */
+  public startPeriodicHealthCheck () {
+    setImmediate(async () => this.healthCheck());
+  }
+
+  private async healthCheck () {
+    console.log('performing health check');
+    if (!this.healthy) {
+      console.log('Unhealthy, restarting IPFS node...');
+      await this.restart();
+      this.healthy = true;
+    } else {
+      console.log('healthy');
+    }
+    setTimeout(async () => this.healthCheck(), this.healthCheckInternalInSeconds * 1000);
```

Comment:
Pattern: consider putting this whole block in try catch-catch, and call `setTimeout()` in `finally` so the monitor never dies.